Source/Core/Core/AchievementManager.cpp

```diff
@@ -148,13 +148,13 @@ void AchievementManager::LoadGameAsync(const DiscIO::Volume* volume,
           },
       .close = [](void* file_handle) { delete reinterpret_cast<FilereaderState*>(file_handle); }};
   rc_hash_init_custom_filereader(&volume_reader);
-  if (!rc_hash_generate_from_file(m_game_hash.data(), RC_CONSOLE_GAMECUBE, ""))
-  {
-    ERROR_LOG_FMT(ACHIEVEMENTS, "Unable to generate achievement hash from game file.");
-    return;
-  }
-  m_loading_volume = nullptr;
   m_queue.EmplaceItem([this, callback] {
+    if (!rc_hash_generate_from_file(m_game_hash.data(), RC_CONSOLE_GAMECUBE, ""))
```

Comment:
Well now you just created a full on race condition. Please think about object lifetimes. You don't know when the volume you store in `m_loading_volume` is going to be destructed.

Reply:
This has not been fixed, for the record.